pkg/distributors/test/RewardsScheduler.test.ts

```diff
@@ -0,0 +1,143 @@
+    );
+  });
+
+  it('emits RewardScheduled', async () => {
+    const time = (await currentTimestamp()).add(3600 * 24);
+    const rewardAmount = fp(1);
+
+    const receipt = await (
+      await rewardsScheduler.connect(rewarder).scheduleReward(pool.address, rewardsToken.address, rewardAmount, time)
+    ).wait();
+
+    const rewardId = await rewardsScheduler.getRewardId(pool.address, rewardsToken.address, rewarder.address, time);
+
+    expectEvent.inReceipt(receipt, 'RewardScheduled', {
+      rewardId,
+      scheduler: rewarder.address,
+      rewardsToken: rewardsToken.address,
+      startTime: time,
+    });
+  });
+
+  it('prevents an unallowlisted rewarder to schedule a reward', async () => {
+    const time = (await currentTimestamp()).add(3600 * 24);
+    const rewardAmount = fp(1);
+    await expect(
+      rewardsScheduler.connect(lp).scheduleReward(pool.address, rewardsToken.address, rewardAmount, time)
+    ).to.be.revertedWith('Only allowlisted rewarders can schedule reward');
+  });
+
+  describe('with a scheduled reward', () => {
+    const rewardAmount = fp(1);
+    let rewardId: string;
+    let time: BigNumber;
+
+    sharedBeforeEach(async () => {
+      // reward duration is important.  These tests assume a very short duration
+      time = (await currentTimestamp()).add(3600 * 24);
+      await rewardsScheduler.connect(rewarder).scheduleReward(pool.address, rewardsToken.address, rewardAmount, time);
+
+      rewardId = await rewardsScheduler.getRewardId(pool.address, rewardsToken.address, rewarder.address, time);
+    });
+
+    it('doesnt reward before time has passed', async () => {
+      await expect(rewardsScheduler.connect(lp).startRewards([rewardId])).to.be.revertedWith(
+        'Reward start time is in the future'
+      );
+    });
+
+    it('responds to getScheduledRewardInfo', async () => {
+      const response = await rewardsScheduler.getScheduledRewardInfo(rewardId);
+
+      expect(response.pool).to.equal(pool.address);
+      expect(response.rewardsToken).to.equal(rewardsToken.address);
+      expect(response.startTime).to.equal(time);
+      expect(response.rewarder).to.equal(rewarder.address);
+      expect(response.amount).to.equal(rewardAmount);
+      expect(response.status).to.equal(1);
+    });
+
+    describe('when time has passed', async () => {
+      sharedBeforeEach(async () => {
+        await advanceTime(3600 * 25);
+      });
+
+      it('allows anyone to poke the contract to notify the staking contract and transfer rewards', async () => {
+        const expectedReward = fp(1);
+
+        await expectBalanceChange(
+          () => rewardsScheduler.connect(lp).startRewards([rewardId]),
+          rewardTokens,
+          [{ account: stakingContract.address, changes: { DAI: ['very-near', expectedReward] } }],
+          vault
+        );
+      });
+
+      it('emits RewardAdded in MultiRewards', async () => {
```

Comment:
We seem to be missing:
 a) a test for the event emitted by the scheduler
 b) a test for what happens once the scheduled reward is started (the status changes to done, it cannot be started again)